frontends/parsers/p4/p4parser.ypp

```diff
@@ -664,12 +666,46 @@ packageTypeDeclaration
 
 instantiation
       : annotations typeRef "(" argumentList ")" name ";"
-                     { $$ = new IR::Declaration_Instance(@6, *$6, new IR::Annotations(*$1),
-                                                         $2, $4);
+                     {
```

Comment:
your implementation is done completely in the bison parser. I would actually suggest constructing an appropriate IR representation of the entire program, including the overridden objects, and then having a compiler pass do this elaboration.

Reply:
The bison implementation is efficient (fastest compiling) by preparing final IR as two programs are being parsed. 
However, for compiling on host machines, code maintenance is key, so why complicate the yacc code?  The bison changes are simple for all, but one case, by checking for a parent before adding an object to IR.  It's only the code that changed to handle `parserState` which is more complicated, but it's only one code object. 

I can document the two new keywords for full scope of their use.  Thereafter, we can decide to use bison or a frontend pass.  If it's a frontend pass, I won't have cycles to work on it anytime soon.

Reply:
> your implementation is done completely in the bison parser. I would actually suggest constructing an appropriate IR representation of the entire program, including the overridden objects, and then having a compiler pass do this elaboration.

Preparing a representation of the entire program in bison for processing later in a frontend pass is easier said than done.  Bison adds IR objects to the symbol table.  The base code is always processed first (modularity entails including base-code.p4) in a P4 program and then writing a new P4 program.  During base code processing, say, `struct foo {}` is added to the symbol table. Then `struct foo {} override` is processed. You cannot add another `foo` to symbol table.  So now bison gets complicated by not adding override and super objects to symbol table.  By the time bison prepares a representation for both programs, most of the bison code in this PR ends up getting used because one has to store the representation of overriden and supered information somewhere.

One could rename `foo` to `foo_ov` during processing override and then add `foo_ov` to symbol table.  But, when the frontend pass processes the representation, the pass needs to remove `foo_ov` from the symbol table.